src/components/Input/Input.tsx

```diff
@@ -30,14 +30,19 @@ interface BaseProps extends Omit<React.InputHTMLAttributes<HTMLInputElement>, 's
     label?: string;
     /** Helper text under the input, display the error when the context is critical */
     helperText?: string;
+    /** Icon before input, optional */
+    leadingIcon?: React.ReactElement;
+    /** Icon after input, which can trigger callback as a button, optional */
+    trailingIcon?: { icon: React.ReactElement; callback: () => void };
     /**  Controls whether space is reserved for error messages under the input field when validation is expected
      *  to avoid "jumping" UI. */
     reserveErrorMessageSpace?: boolean;
 }
 
-export type Props = BaseProps & (ErrorStateProps | { context?: undefined; errorMessage?: never });
+export type Props = BaseProps &
+    (ErrorStateProps | { context?: undefined | string; errorMessage?: undefined | string });
```

Comment:
Because it could be that the error is passed to the SearchInput as a prop, i thought. And in that case it has to be passed on to Input... 
I was in doubt when I did this though ;) I thought it was a bit strange anyway, that it is set to undefined or never, but you also have this:
type ErrorStateProps = {
    /** The input error field context (critical) */
    context: ErrorContext;
    /** This message will be rendered under the input when context critical will be applied */
    errorMessage: string;
};

I'm not familiar with this structure!

Reply:
the idea critical part here is that you have `ErrorStateProps | {...}`. This means you either pass the ErrorStateProps which will defined those details, or you don't pass neither. So in this way we group props into a logical unit that means if context is undefined, errorMessage will also not be there. Something like that. Does that clarify things?
@AnastasiiaMP came up with this pattern recently, I think we used in it a 2-3 components now.

With your change, this logic is overwritten, as you allow for the error message to be there even in the second option. so this change should be reverted.